### PR TITLE
Simplify status compression

### PR DESCRIPTION
**Pull request by tigrannajaryan**
- Eliminate all "hash" fields.
- Allow omitting the sub-messages of AgentToServer message. When omitted it is implied that previously reported value of the sub-message is current (unchanged).
- To detect lost messages have one auto-incremented sequence_num field AgentToServer message. Server can easily detect losses by just keeping the last sequence_num (as opposed to keeping 4 different hashes).

### CODE REVIEW

**Comment by andykellr** [comment]
Actually, doing a search for `Hash` I found a couple more places that should be updated. I'll add suggestions.

File: client/internal/clientcommon.go [hunk 1]
```diff
@@ -310,15 +270,14 @@ func (c *ClientCommon) SetPackageStatuses(statuses *protobufs.PackageStatuses) e
...
 
-	// Get the hash of the status before we update it.
-	prevHash := c.ClientSyncedState.PackageStatuses().GetHash()
+	statusChanged := !proto.Equal(c.ClientSyncedState.PackageStatuses(), statuses)
 
 	if err := c.ClientSyncedState.SetPackageStatuses(statuses); err != nil {
 		return err
 	}
 
 	// Check if the new status is different from the previous by comparing the hashes.
```
Comment (andykellr) [inline client/internal/clientcommon.go:279]:
```suggestion
	// Check if the new status is different from the previous.
```
```suggestion
	// Check if the new status is different from the previous by comparing the hashes.
```

**Comment by andykellr** [comment]
The following have comments that refer to the Hash field:
* client/types/startsettings.go
* client/types/callbacks.go
* client/client.go

**Comment by tigrannajaryan** [comment]
> The following have comments that refer to the Hash field:
> 
> * client/types/startsettings.go
> * client/types/callbacks.go
> * client/client.go

Thanks for the catch. I did a pass over this files. Please check again.

File: client/internal/clientcommon.go (hunk 1 above)
Comment (tigrannajaryan) [reply client/internal/clientcommon.go]: Fixed.

**Pull request merged**